src/api/c/imageio2.cpp

```diff
@@ -61,9 +61,9 @@ static af_err readImage_t(af_array *rImage, const uchar* pSrcLine, const int nSr
                 } else {
                     // Non 8-bit types do not use ordering
                     // See Pixel Access Functions Chapter in FreeImage Doc
-                    pDst0[indx] = (T) *(src + (x * step + 0));
-                    pDst1[indx] = (T) *(src + (x * step + 1));
-                    pDst2[indx] = (T) *(src + (x * step + 2));
+                    pDst0[indx] = (T) *(src + (x * step + FI_RGBA_RED));
+                    pDst1[indx] = (T) *(src + (x * step + FI_RGBA_GREEN));
+                    pDst2[indx] = (T) *(src + (x * step + FI_RGBA_BLUE));
```

Comment:
No imageio.cpp is fine. I checked your PR when you did it. I check this file as well. Somehow the u8 code skipped my eye.
This only affects 16-bit and 32-bit images. That's why u8 is using the enums in the if(u8) case above. That's why imageio.cpp should be fine.

Essentially the 4xu8 are 32-bits, so they keep that as one unit. But for 16 and 32-bit it doesn't matter, so they don't use enums.